src/ProductBundle/Controller/BaseProductController.php

```diff
@@ -89,9 +90,11 @@ public function renderPropertiesAction(ProductInterface $product)
      */
     public function renderFormBasketElementAction(FormView $formView, BasketElementInterface $basketElement, BasketInterface $basket)
     {
-        $provider = $this->get('sonata.product.pool')->getProvider($basketElement->getProduct());
+        /** @var Pool $pool */
+        $pool = $this->get('sonata.product.pool');
```

Comment:
Please add a constructor with one optional argument. If the argument is provided, use it, and if not, please trigger a deprecation error and default to this.

Reply:
There are various other places in this controller that this service is used, and I don't want to make too many unrelated changes to this PR. So should only this instance be updated, or can this stay as-is for now then everything can be updated in on go in a separate PR?

Reply:
It can stay as is, I didn't know about the other occurences 👍